libraries/chain/finality/finalizer.cpp

```diff
@@ -85,6 +85,19 @@ finalizer::vote_result finalizer::decide_vote(const block_state_ptr& bsp) {
    return res;
 }
 
+// ----------------------------------------------------------------------------------------
+// finalizer has voted strong on bsp, update finalizer safety info if more recent than the current lock
+bool finalizer::maybe_update_fsi(const block_state_ptr& bsp) {
+   auto& final_on_strong_qc_block_ref = bsp->core.get_block_reference(bsp->core.final_on_strong_qc_block_num);
+   if (final_on_strong_qc_block_ref.timestamp > fsi.lock.timestamp) {
```

Comment:
I think we should do the update iff `fsi,lock.empty()`, so not compare the timestamp.

Reply:
I understood the requirement to be if more recent. @arhag

Reply:
Yes, you are right, my bad.